MetaMask/metamask-extension
# Degrade gracefully when gas API is down

When the gas API is down, the logic we use will no longer compute all of
the data that the gas API returns in order to reduce the burden on
Infura. Specifically, only estimated fees for different priority levels,
as well as the latest base fee, will be available; all other data
points, such as the latest and historical priority fee range and network
stability, will be missing. This commit updates the frontend logic to
account for this lack of data by merely hiding the relevant pieces of
the UI that would otherwise be shown.

---

## Related issues

* https://github.com/MetaMask/controllers/pull/712

## Manual testing steps

* Open the MetaMask popup.
* Go to send a transaction.
* Proceed to the confirm transaction screen.
* Click on "Market" (or whatever priority level is currently selected).
* Note that all of the data under Network Status should be present.
* Click on "Advanced".
* Note that all of the information under each field should be present ("current" and "12hr").
* Close the Advanced modal.
* Find the extension in your browser and open up the inspector for `background.html`.
* Go to the Network tab.
* Find a request that looks like `suggestedGasFees`. Right-click on it and choose "Block request URL".
* Open the MetaMask popup again.
* Note that 2 out of the 3 sections in Network Status should be gone (only Base Fee should be there).
* Open Advanced.
* Note that most of the information under each field should be gone (only "current" should be there).

## Screencaps

**When gas API is up:**

This should be the same behavior as today.

https://user-images.githubusercontent.com/7371/157169150-d55fd1df-8c01-45e9-8e24-68dd1b905730.mov

**When gas API is down:**

Note that the number of requests stays the same, i.e. there shouldn't be excessive `591f0dce...` lines in the request log, and specifically there shouldn't be excessive requests that call `eth_feeHistory`. Also note that some data disappears from the gas and advanced gas fee modals.

https://user-images.githubusercontent.com/7371/157169187-8c33ab9d-72ca-488d-849d-137b56c432fa.mov
